fvm/state/ledger_address_generator.go

```diff
@@ -11,18 +13,18 @@ const keyAddressState = "account_address_state"
 // The only change is that when next address is called the state is updated as well.
 type LedgerBoundAddressGenerator struct {
 	generator flow.AddressGenerator
-	ledger    Ledger
+	state     *State
 }
 
-func NewLedgerBoundAddressGenerator(ledger Ledger, chain flow.Chain) (*LedgerBoundAddressGenerator, error) {
-	stateBytes, err := ledger.Get("", "", keyAddressState)
+func NewLedgerBoundAddressGenerator(state *State, chain flow.Chain) (*LedgerBoundAddressGenerator, error) {
```

Comment:
Would it now be StateBoundAddressGenerator :) that would make more sense anyway.

Reply:
Agree!

Reply:
applied at fd3ec2a